docs/source/cuda/memory.rst

```diff
@@ -156,6 +156,94 @@ traditional dynamic memory management.
 .. seealso::
    :ref:`Matrix multiplication example <cuda-matmul>`.
 
+Dynamic Shared Memory
+---------------------
+
+In order to use dynamic shared memory in kernel code declare a shared array of
+size 0:
+
+.. code-block:: python
+
+   @cuda.jit
+   def kernel_func(x):
+      dyn_arr = cuda.shared.array(0, dtype=np.float32)
+      ...
+
+and specify the size of dynamic shared array in bytes during kernel invocation:
```

Comment:
Sorry, I missed this last night - I'd suggest we're specifying the size of dynamic shared memory, rather than an array, to be consistent with the terminology we're generally using when talking about the declaration:
```suggestion
and specify the size of dynamic shared memory in bytes during kernel invocation:
```